expression.go

```diff
@@ -81,6 +81,24 @@ func (expr ValueExpr) RootRefs() []FieldExpr {
 	return nil
 }
 
+type NotExpr []Expression
```

Comment:
not expression always has only one subexpr. is it necessary to use an array here?

Reply:
@ysui6888 You should take a look at the implementation here which I believe @nelio2k is going to use to superseed his NotExpr implementation:
https://github.com/couchbaselabs/gojsonsm/pull/11/commits/b9c7abf2e9255180077cd67b1222ebf9507f5607